### ISSUE TITLE
header/core: move `Listener` and `Exchange` out to the root `core` pkg

### ISSUE THREAD

**Issue by Wondertan** [celestiaorg/celestia-node#1424]
The listener works with block data and constructs EDSes, but it is located under the `header` pkg, which should not depend on `share` anyhow. Further, the `Listener` will depend on `eds.Store`(#1423) which may introduce a dep cycle.

The same is true for the `header/core.Exchange`

**Comment by Wondertan** [comment]
This will likely be done by #1304

**Comment by tzdybal** [comment]
You want me to do it? Or you're going to refactor after #1304?

**Comment by Wondertan** [comment]
@tzdybal, I was thinking about this issue from a different perspective but realized that your PR touches on it as well and moves those components anyway, so would be nice to move them to the root `core` pkg

**Comment by renaynay** [comment]
Already done.
